File: Sliding_Window/121_best_time_to_buy_and_sell_stock/121.cpp
#include "iostream";
#include "vector";

int maxProfits(std::vector<int> &prices) {

    int min_price = prices[0];
    int max_profits = 0;

    for (int i = 0; i < prices.size(); i++) {
        min_price = std::min(prices[i], min_price);
        max_profits = std::max(prices[i] - min_price, max_profits);
    }
    return max_profits;
}

int main() {

    std::vector<int> prices = {7,1,5,3,6,4};
    std::cout << "result : " << maxProfits(prices) << std::endl; 

    return 0;
}

File: Sliding_Window/121_best_time_to_buy_and_sell_stock/121.rs


fn max_profit(prices: Vec<i32>) -> i32 {
    let mut min_price: i32 = prices[0];
    let mut max_profits: i32 = 0;

    for i in 1..prices.len() {
        min_price = std::cmp::min(prices[i], min_price);
        max_profits = std::cmp::max(prices[i] - min_price, max_profits);
    }

    return max_profits;
}

fn main() {
    
    let prices: Vec<i32> = vec![7,1,5,3,6,4];

    println!("result : {}", max_profit(prices));

}